realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -727,21 +757,44 @@ public void setObject(String fieldName, @Nullable DynamicRealmObject value) {
      * Sets the reference to a {@link RealmList} on the given field.
      *
      * @param fieldName field name.
-     * @param list list of references.
-     * @throws IllegalArgumentException if field name doesn't exist, it is not a list field, the type
-     * of the object represented by the DynamicRealmObject doesn't match or any element in the list belongs to a
-     * different Realm.
+     * @param list list of objects. Must either be primitive types or Realm objects.
```

Comment:
This should be a bit more specific that it requires `{@link DynamicRealmObject}`.

Reply:
Done